StaticGUI.py

```diff
@@ -154,15 +154,61 @@ def runSimulation(self, smoothPath):
 class SmoothedPathGUI(MapPathGUI):
     def __init__(self, master, inputMap, path):
         super().__init__(master, inputMap, path)
+        self.prev_line_id = []
+        self.set_of_prev_path = []
 
     def drawPath(self):
+        # change previous 5 path to rainbow colors
+
+        color = 'gray1'
+        if self.prev_line_id:
+            self.set_of_prev_path.append(self.prev_line_id)
+
+        if self.set_of_prev_path:
```

Comment:
Maybe you can put the colors in a list called colors and then shorten these lines of code to a for loop. Replace the indexes inside self.set_of_prev_path with the variable in the for loop (i.e the variable i for example) and call colors[i] inside the if statement.

Reply:
How about this....
color = None
colors = ['#347800', '#347800'...]
if(self.prev_line_id in self.set_of_orev_path):
    color = colors[self.set_of_orev_path.index(self.prev_line_id)

Reply:
I'd also make the colors lists something you put in consts.py if you use this. All the wonky hashtags should be given word names like GREEN0, GREEN1... or something intuitive